calico-vpp-agent/routing/bgp_watcher.go

```
@@ -267,6 +269,180 @@ func (w *Server) startBGPMonitoring() (func(), error) {
 	return stopFunc, err
 }
 
+func (w *Server) NewBGPPolicyAndAssignment(name string, rules []calicov3.BGPFilterRuleV4, neighborName string, dir bgpapi.PolicyDirection) (*watchers.BGPPolicyAndAssignment, error) {
+	pol := &bgpapi.Policy{Name: name}
+	for _, rule := range rules {
+		routeAction := bgpapi.RouteAction_ACCEPT
+		if rule.Action == calicov3.Reject {
+			routeAction = bgpapi.RouteAction_REJECT
+		} else if rule.Action != calicov3.Accept {
+			w.log.Errorf("action %s not supported", rule.Action)
+		}
+
+		var matchSetType bgpapi.MatchSet_Type
+		var minMask, maxMask uint32
+		if rule.MatchOperator == calicov3.In || rule.MatchOperator == calicov3.NotIn {
+			_, subnet, err := net.ParseCIDR(rule.CIDR)
+			if err != nil {
+				w.log.Error(err)
```

Same here, we should probably return an error